src/main/java/com/opencart/pages/HeaderPage.java

```diff
@@ -2,19 +2,26 @@
 
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 
 public class HeaderPage extends BasePage {
 
     @FindBy(xpath = ".//*[@title = 'My Account']")
     private WebElement myAccountButton;
     @FindBy(xpath = ".//*[contains(@href,'register')]")
     private WebElement registerButton;
+    @FindBy(xpath = ".//*[contains(@href,'login')]" )
+    private WebElement loginButton;
 
     public WebElement getMyAccountButton() {
+        wait.until(ExpectedConditions.visibilityOf(myAccountButton));
         return myAccountButton;
     }
-
     public WebElement getRegisterButton() {
         return registerButton;
     }
+    public WebElement getLoginButton(){
+        wait.until(ExpectedConditions.visibilityOf(loginButton));
```

Comment:
try to use not only visibilityOf for each element, for the button you can put expected condition e.g elementToBeClickable​.
 Make sure an element is visible and enabled such that you can click it.

https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/ui/ExpectedConditions.html#elementToBeClickable(org.openqa.selenium.By)